src/overview/actions.js

```diff
@@ -258,11 +265,19 @@ export const showTags = index => (dispatch, getState) => {
 
 export const filterTag = tag => (dispatch, getState) => {
     const query = selectors.query(getState())
+    const filterTags = selectors.filterTags(getState())
+
     const transformedTag = `#${tag.split(' ').join('+')} `
 
-    const newQuery = query.includes(transformedTag)
+    const newQuery = query.includes(transformedTag.slice(0, -1))
```

Comment:
Now there is a dedicated tags filter input + corresponding state, I don't think pill clicks should mutate the query state anymore - this was more of a quick hack to get it working without a tag filter UI, and things will obviously get very difficult if we try to keep the two states synced. `filterTags` can be used to manage all tags filter state, but the simple `#tag` parsing can still exist in query for user convenience (happens at backend anyway - not UI-related). Would be good to open the filters menu whenever the `filterTags` state is changed, so user gets feedback that the state has change. 
@oliversauter what do you think should happen when a pill is clicked (on one of the results, not new filters)?

Reply:
I expect the same thing to happen as now. 
At least from a user POV clicking the pill and applying a filter is not a quick hack, its expected behaviour.

Reply:
Seems fairly complicated but @mukeshkharita seem to have made a good start. I think we need to define the corresponding state change behaviours for the actions:
1. click on pills in results
2. type "#tag" patterns in query
3. start backspacing over "#tag" patterns in query
4. delete pills from filter

1 currently works by placing the corresponding "#tag" pattern of that pill into the query state + adding tag to tags filter state. Seems expected.

2 currently detects the "#tag" pattern in the query and adds it to the tags filter state. seems like it works

3 4 currently works by changing the tag in the filter state. 
eg: 
query state:`#tags`
filter state: `['tags']`
* backspace over the 's'
query state:`#tag`
filter state: `['tag']`
selecting the entire query and backspacing breaks the sync by not updating the filter state though.

4 doesn't seem to be implemented. This breaks sync because if you manually delete a tag from the filter, the query state will remain the same, and as soon as it's changed next, the tag will be re-added to filter state (because of 2)

I think 2, 3 and 4 apply for domains filter as well (2 + 3 implemented). Any other actions?